File: arrays/array_variables-two_dimensional_arrays.py
names = [
    ['John', 'Jane', 'Jack', 'Jill', 'Jenny', 'Jerry'],
    ['Blue', 'Black', 'Cat', 'Marik', 'Joel', 'Doe']
]

# Linear search algorithm
def linear_search(arr, value):
    for index, item in enumerate(arr):
        if item == value:
            return index
        for i, row in enumerate(item):
            if row == value:
                return (index, i)
    return -1

print(f"Result for search: {linear_search(names, 'Jill')}")  #(0, 3)

# enum = enumerate(names)
# for indice, col in enum:
#     if col == 'Jill':
#         print(indice)
#     for i, row in enumerate(col):
#         if row == 'Jill':
#             print((indice,i))

File: data-structures/01-arrays/array_variables-two_dimensional_arrays.js
let names = [
  ["John", "Jane", "Jack", "Jill", "Jenny", "Jerry"],
  ["Blue", "Black", "Cat", "Marik", "Joel", "Doe"],
];


function linear_search(arr, item) {
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr[i].length; j++) {
      if (arr[i][j] === item) {
        return [i, j];
      }
    }
  }
  return -1;  // if the item is not found
}

console.log(linear_search(names, "Jill")); // [0, 3]